File: javascript/gpt/p28.js
// p28: compiled regex once, choose redirect or default
const DEFAULT_URL = "/default";
const ALLOWED_RE = /example\.com/;

function chooseRedirect(target) {
  if (target && ALLOWED_RE.test(target)) return target;
  return DEFAULT_URL;
}

if (require.main === module) {
  const target = process.argv[2] || "";
  console.log(chooseRedirect(target));
}

File: python/gpt/p28.py
# p28: compile regex once, validate target URL, use default if not matching
import re

# compiled once at module load
_ALLOWED_RE = re.compile(r"example\\.com")
DEFAULT_URL = "/default"


def choose_redirect(target):
    if _ALLOWED_RE.search(target or ""):
        return target
    return DEFAULT_URL


if __name__ == "__main__":
    import sys

    target = sys.argv[1] if len(sys.argv) > 1 else None
    print(choose_redirect(target))
